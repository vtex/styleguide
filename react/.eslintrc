{
  "extends": "vtex-react",
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  }
}
