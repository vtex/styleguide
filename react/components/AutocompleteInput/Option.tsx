import classNames from 'classnames'
import PropTypes from 'prop-types'
import React, { useState } from 'react'

/** Structured Autocomplete option shape */
const structuredAutocompleteOptionShape = PropTypes.shape({
  value: PropTypes.any,
  label: PropTypes.string.isRequired,
})

type StructuredAutocompleteOption = PropTypes.InferType<
  typeof structuredAutocompleteOptionShape
>

export type AutocompleteOption = string | StructuredAutocompleteOption

/** Autocomplete option shape */
export const autocompleteOptionShape = PropTypes.oneOfType([
  PropTypes.string,
  structuredAutocompleteOptionShape,
])

export const getTermFromOption = (option: AutocompleteOption): string =>
  typeof option === 'string' ? option : option.label

const propTypes = {
  /** Determine if the option should have a rounded bottom */
  roundedBottom: PropTypes.bool,
  /** Option icon */
  icon: PropTypes.node,
  /** Search term used to highlight it into the label */
  searchTerm: PropTypes.string.isRequired,
  /** Option title */
  value: autocompleteOptionShape.isRequired,
  /** Determine if an option is selected and should be highlighted */
  selected: PropTypes.bool.isRequired,
  /** Click handler */
  onClick: PropTypes.func.isRequired,
}

const Option: React.FunctionComponent<PropTypes.InferProps<
  typeof propTypes
>> = props => {
  const { icon, selected, roundedBottom, searchTerm, onClick } = props
  const [highlightOption, setHighlightOption] = useState(false)
  const value = getTermFromOption(props.value)

  const renderOptionValue = (): string | React.ReactElement => {
    const index = value.toLowerCase().indexOf(searchTerm.toLowerCase())
    if (index === -1) {
      return value
    }
    const prefix = value.substring(0, index)
    const match = value.substr(index, searchTerm.length)
    const suffix = value.substring(index + match.length)
    return (
      <span className="truncate">
        <span className="fw7">{prefix}</span>
        {match}
        <span className="fw7">{suffix}</span>
      </span>
    )
  }

  const buttonClasses = classNames('bn w-100 tl pointer pa4 f6', {
    'br2 br--bottom': roundedBottom,
    'bg-muted-5': highlightOption || selected,
    'bg-base': !highlightOption && !selected,
  })

  return (
    <button
      className={buttonClasses}
      onFocus={() => setHighlightOption(true)}
      onMouseEnter={() => setHighlightOption(true)}
      onMouseLeave={() => setHighlightOption(false)}
      onClick={onClick}
    >
      <span className="h1 flex items-center">
        <span className="mr3 c-muted-2 flex pt1">{icon}</span>
        {renderOptionValue()}
      </span>
    </button>
  )
}

Option.propTypes = propTypes

export default Option
