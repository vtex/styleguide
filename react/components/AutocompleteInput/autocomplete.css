.mh300 {
  max-height: 300px;
}

.scroll {
  overflow: scroll;
}
