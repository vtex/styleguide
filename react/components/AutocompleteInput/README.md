#### AutocompleteInput is a search input which, enhanced with a suggestions pop-up, gives the user a more complete search experience when needed.

### 👍 Dos

- Help the user avoid errors.
- Semantically similar phrases or words but grammarly different should appear once. E.g.: "Dress" and "Dresses".
- Highlight the suggestions.

### 👎 Don'ts

- Exceed 10 suggestions. 6 is a great number of suggestions.

#### Simple Autocomplete mocking server responses

```jsx
import { uniq } from 'lodash'
import { useState, useRef } from 'react'

const allUsers = [
  'Ana Clara',
  'Ana Luiza',
  { value: 1, label: 'Bruno' },
  'Carlos',
  'Daniela',
]

const UsersAutocomplete = () => {
  const [term, setTerm] = useState('')
  const [loading, setLoading] = useState(false)
  const timeoutRef = useRef(null)

  const options = {
    onSelect: (...args) => console.log('onSelect: ', ...args),
    loading,
    value: !term.length
      ? []
      : allUsers.filter(user =>
          typeof user === 'string'
            ? user.toLowerCase().includes(term.toLowerCase())
            : user.label.toLowerCase().includes(term.toLowerCase())
        ),
  }

  const input = {
    onChange: term => {
      if (term) {
        setLoading(true)
        if (timeoutRef.current) {
          clearTimeout(timeoutRef.current)
        }
        timeoutRef.current = setTimeout(() => {
          setLoading(false)
          setTerm(term)
          timeoutRef.current = null
        }, 1000)
      } else {
        setTerm(term)
      }
    },
    onSearch: (...args) => console.log('onSearch:', ...args),
    onClear: () => setTerm(''),
    placeholder: 'Search user... (e.g.: Ana)',
    value: term,
  }
  return <AutocompleteInput input={input} options={options} />
}

;<UsersAutocomplete />
```

#### Show search history

```jsx
import { uniq } from 'lodash'
import { useState, useRef } from 'react'

const allUsers = [
  'Ana Clara',
  'Ana Luiza',
  { value: 1, label: 'Bruno' },
  'Carlos',
  'Daniela',
]

const UsersAutocomplete = () => {
  const [term, setTerm] = useState('')
  const [loading, setLoading] = useState(false)
  const [lastSearched, setLastSearched] = useState([])
  const timeoutRef = useRef(null)

  const options = {
    onSelect: (...args) => console.log('onSelect: ', ...args),
    loading,
    value: !term.length
      ? []
      : allUsers.filter(user =>
          typeof user === 'string'
            ? user.toLowerCase().includes(term.toLowerCase())
            : user.label.toLowerCase().includes(term.toLowerCase())
        ),
    // --- This is what makes the Last Searched Terms work!
    // This can be stored anywhere the dev wants. To be persistent, for example.
    lastSearched: {
      value: lastSearched,
      label: 'Last searched users',
      onChange: option =>
        option && setLastSearched(uniq([...lastSearched, option])),
    },
  }

  const input = {
    onChange: term => {
      if (term) {
        setLoading(true)
        if (timeoutRef.current) {
          clearTimeout(timeoutRef.current)
        }
        timeoutRef.current = setTimeout(() => {
          setLoading(false)
          setTerm(term)
          timeoutRef.current = null
        }, 1000)
      } else {
        setTerm(term)
      }
    },
    onSearch: (...args) => console.log('onSearch:', ...args),
    onClear: () => setTerm(''),
    placeholder: 'Search user... (e.g.: Ana)',
    value: term,
  }
  return <AutocompleteInput input={input} options={options} />
}

;<UsersAutocomplete />
```

#### Custom option rendering

```jsx
import { uniq } from 'lodash'
import { useState, useRef } from 'react'

import User from '../icon/User'
import Search from '../icon/Search'

const allUsers = [
  'Ana',
  { value: 1, label: 'Ana Clara' },
  { value: 2, label: 'Ana Luiza', caption: 'VTEX' },
  { value: 3, label: 'Bruno' },
]

const CustomOption = props => {
  const { roundedBottom, searchTerm, value, selected, onClick } = props
  const [highlightOption, setHighlightOption] = useState(false)

  const renderOptionHighlightedText = () => {
    const highlightableText = typeof value === 'string' ? value : value.label
    const index = highlightableText
      .toLowerCase()
      .indexOf(searchTerm.toLowerCase())
    if (index === -1) {
      return highlightableText
    }
    const prefix = highlightableText.substring(0, index)
    const match = highlightableText.substr(index, searchTerm.length)
    const suffix = highlightableText.substring(index + match.length)
    return (
      <span className="truncate">
        <span className="fw7">{prefix}</span>
        {match}
        <span className="fw7">{suffix}</span>
      </span>
    )
  }

  const buttonClasses = `bn w-100 tl pointer pa4 f6 ${
    roundedBottom ? 'br2 br--bottom' : ''
  } ${highlightOption || selected ? 'bg-muted-5' : 'bg-base'}`

  const icon = typeof value === 'string' ? <Search size={14} /> : <User size={14} />

  // --- This is a good practice. Use <button />.
  return (
    <button
      className={buttonClasses}
      onFocus={() => setHighlightOption(true)}
      onMouseEnter={() => setHighlightOption(true)}
      onMouseLeave={() => setHighlightOption(false)}
      onClick={onClick}>
      <div className="flex items-center">
        <span className="mr3 c-muted-2 flex pt1">{icon}</span>
        <span className="pr2">{renderOptionHighlightedText()}</span>
        {typeof value !== 'string' && (
          <div className="t-mini c-muted-1">{value.caption}</div>
        )}
      </div>
    </button>
  )
}

const UsersAutocomplete = () => {
  const [term, setTerm] = useState('')
  const [loading, setLoading] = useState(false)
  const [lastSearched, setLastSearched] = useState([])
  const timeoutRef = useRef(null)

  const options = {
    onSelect: (...args) => console.log('onSelect: ', ...args),
    loading,
    value: !term.length
      ? []
      : allUsers.filter(user =>
          typeof user === 'string'
            ? user.toLowerCase().includes(term.toLowerCase())
            : user.label.toLowerCase().includes(term.toLowerCase())
        ),
    lastSearched: {
      value: lastSearched,
      label: 'Last searched users',
      onChange: option =>
        option && setLastSearched(uniq([...lastSearched, option])),
    },
    // --- This is what makes the custom option work!
    renderOption: props => <CustomOption {...props} />,
  }

  const input = {
    onChange: term => {
      if (term) {
        setLoading(true)
        if (timeoutRef.current) {
          clearTimeout(timeoutRef.current)
        }
        timeoutRef.current = setTimeout(() => {
          setLoading(false)
          setTerm(term)
          timeoutRef.current = null
        }, 1000)
      } else {
        setTerm(term)
      }
    },
    onSearch: (...args) => console.log('onSearch:', ...args),
    onClear: () => setTerm(''),
    placeholder: 'Search user... (e.g.: Ana)',
    value: term,
  }
  return <AutocompleteInput input={input} options={options} />
}

;<UsersAutocomplete />
```
