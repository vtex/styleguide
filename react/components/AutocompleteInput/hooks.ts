import { useState, useEffect } from 'react'

export const useClickOutside = (
  ref: React.RefObject<HTMLElement> | null,
  onClickOutside: (e: Event) => void
) => {
  const handleClickOutside = (e: Event) => {
    if (
      ref?.current &&
      e.target instanceof Node &&
      !ref.current.contains(e.target)
    ) {
      onClickOutside(e)
    }
  }

  useEffect(() => {
    // Bind the event listener
    document.addEventListener('mousedown', handleClickOutside)
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener('mousedown', handleClickOutside)
    }
  })
}

export const useArrowNavigation = (
  ref: React.RefObject<HTMLElement> | null,
  optionsLength: number,
  initialSelectedOptionIndex: number
): [number, (index: number) => void] => {
  const [selectedOptionIndex, setSelectedOptionIndex] = useState(
    initialSelectedOptionIndex
  )

  const handleKeyDown = (e: KeyboardEvent) => {
    if (
      ref?.current &&
      e.target instanceof Node &&
      ref.current.contains(e.target)
    ) {
      let index
      switch (e.key) {
        case 'ArrowUp':
          e.preventDefault()
          index = Math.max(selectedOptionIndex - 1, initialSelectedOptionIndex)
          return setSelectedOptionIndex(index)
        case 'ArrowDown':
          e.preventDefault()
          index = Math.min(selectedOptionIndex + 1, optionsLength - 1)
          return setSelectedOptionIndex(index)
        default:
      }
    }

    return null
  }

  useEffect(() => {
    // Bind the event listener
    document.addEventListener('keydown', handleKeyDown)
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener('keydown', handleKeyDown)
    }
  })

  return [selectedOptionIndex, setSelectedOptionIndex]
}
