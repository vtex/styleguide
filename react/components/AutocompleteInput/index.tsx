import PropTypes from 'prop-types'
import React, { useState, useRef } from 'react'

import Spinner from '../Spinner'
import { useClickOutside, useArrowNavigation } from './hooks'
import Option, {
  AutocompleteOption,
  autocompleteOptionShape,
  getTermFromOption,
} from './Option'
import SearchInput from './SearchInput'

const propTypes = {
  /** Input props. All HTMLInput props can be added too */
  input: PropTypes.shape({
    /** Clear input handler */
    onClear: PropTypes.func.isRequired,
    /** Search by term handler (fired on enter or when clicking the search button) */
    onSearch: PropTypes.func.isRequired,
    /** Change term handler */
    onChange: PropTypes.func.isRequired,
    /** Term to be searched */
    value: PropTypes.string,
    /** Determine if the input and button should be disabled */
    disabled: PropTypes.bool,
  }).isRequired,
  /** Options props. More details in the examples */
  options: PropTypes.shape({
    /**
     * Determine if a spinner will be shown below the given options
     * to show that more options will be added
     */
    loading: PropTypes.bool,
    /**
     * Function that makes possible to the dev to customly render option.
     * Called with all props needed: `(props: { key: string, selected: boolean, value: OptionValue, searchTerm: string, roundedBottom: boolean, icon: ReactElement, onClick: () => void }, index: number)` and should return a React Node
     */
    renderOption: PropTypes.func,
    /**
     * List of options.
     * An option could be a string (denoting a search by term) or an object
     * with `{value: any, label: string}` (denoting the search is related to an entity).
     */
    value: PropTypes.arrayOf(autocompleteOptionShape).isRequired,
    /**
     * Icon representing the entity.
     * Shown when a value is an object to show the difference
     */
    icon: PropTypes.element,
    /**
     * Callback called when an option is selected
     * (clicked or via arrow keys + enter)
     */
    onSelect: PropTypes.func.isRequired,
    /**
     * Last searched terms. Can be used to enhance the Autocomplete experience.
     * Defined with: `{
     *   value: OptionValue[],
     *   onChange: (term: string | OptionValue) => any,
     *   label: string
     * }`
     */
    lastSearched: PropTypes.shape({
      /** List of last searched options */
      value: PropTypes.arrayOf(autocompleteOptionShape).isRequired,
      /**
       * Last searched change handler.
       * Called when a term is searched or an option is selected.
       */
      onChange: PropTypes.func,
      /** Last Searched options's title */
      label: PropTypes.node.isRequired,
    }),
  }).isRequired,
}

export type AutocompleteInputProps = PropTypes.InferProps<typeof propTypes>

const AutocompleteInput: React.FunctionComponent<PropTypes.InferProps<
  typeof propTypes
>> = ({
  input: { value, onClear, onSearch, onChange, ...inputProps },
  options: {
    onSelect,
    value: options,
    renderOption,
    loading,
    lastSearched = {},
    icon,
  },
}) => {
  const [term, setTerm] = useState(value || '')
  const [showPopover, setShowPopover] = useState(false)
  const containerRef = useRef<HTMLDivElement>(null)
  const searching = term.length
  const showLastSearched =
    !searching && lastSearched.value && lastSearched.value.length > 0

  const getShowedOptions = (): AutocompleteOption[] => {
    if (showLastSearched) {
      return lastSearched.value
    }
    if (searching) {
      return options
    }
    return []
  }

  const showedOptions = getShowedOptions()

  const noSelectedOptionIndex = -1
  const [selectedOptionIndex, setSelectedOptionIndex] = useArrowNavigation(
    containerRef,
    showedOptions.length,
    noSelectedOptionIndex
  )
  useClickOutside(containerRef, () => setShowPopover(false))

  const addToLastSearched = (option: AutocompleteOption) => {
    if (lastSearched && lastSearched.onChange) {
      lastSearched.onChange(option)
    }
  }

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      const selectedOption =
        selectedOptionIndex !== -1 ? showedOptions[selectedOptionIndex] : term
      addToLastSearched(selectedOption)
      setTerm(getTermFromOption(selectedOption))
      if (selectedOptionIndex !== -1) {
        onSelect(selectedOption)
      } else {
        onSearch(getTermFromOption(selectedOption))
      }
      setSelectedOptionIndex(-1)
      setShowPopover(false)
    }
  }

  const handleTermChange = (newTerm = '') => {
    if (!showPopover) {
      setShowPopover(true)
    }
    setTerm(newTerm)
    if (onChange) {
      onChange(newTerm)
    }
    setSelectedOptionIndex(noSelectedOptionIndex)
  }

  const handleClear = () => {
    setSelectedOptionIndex(noSelectedOptionIndex)
    setShowPopover(false)

    setTerm('')
    onClear()
  }

  const handleOptionClick = (option: AutocompleteOption) => {
    setTerm(getTermFromOption(option))
    onSelect(option)
    setShowPopover(false)
  }

  const getOptionProps = (option, index) => ({
    key: `${getTermFromOption(option)}-${index}`,
    selected: index === selectedOptionIndex,
    value: option,
    searchTerm: term,
    roundedBottom: index === showedOptions.length - 1,
    icon: typeof option !== 'string' && icon ? icon : null,
    onClick: () => {
      addToLastSearched(option)
      handleOptionClick(option)
    },
  })

  const renderOptions = (): React.ReactElement | React.ReactElement[] => (
    <div className="flex flex-column">
      {showLastSearched ? (
        <div className="pa4 b f6">
          {lastSearched.label || 'Last searched terms'}
        </div>
      ) : null}
      {showedOptions.map((option, index) =>
        renderOption ? (
          renderOption(getOptionProps(option, index), index)
        ) : (
          <Option {...getOptionProps(option, index)} />
        )
      )}
    </div>
  )

  const popoverOpened = showPopover && (!!showedOptions.length || loading)

  return (
    <div ref={containerRef} className="flex flex-column w-100">
      <SearchInput
        {...inputProps}
        value={
          selectedOptionIndex === -1
            ? term
            : getTermFromOption(showedOptions[selectedOptionIndex])
        }
        roundedBottom={!popoverOpened}
        onKeyDown={handleKeyDown}
        onFocus={() => setShowPopover(true)}
        onSearch={() => onSearch(term)}
        onClear={handleClear}
        onChange={handleTermChange}
      />
      {popoverOpened ? (
        <div className="relative">
          <div className="absolute br--bottom br2 bb bl br bw1 b--muted-3 bg-base w-100 z-1 shadow-5">
            {renderOptions()}
            {loading && (
              <div className="flex flex-row justify-center items-center pa4">
                <Spinner size={20} />
              </div>
            )}
          </div>
        </div>
      ) : null}
    </div>
  )
}

AutocompleteInput.propTypes = propTypes

export default AutocompleteInput
