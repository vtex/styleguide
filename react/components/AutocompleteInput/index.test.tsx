import React from 'react'
import { render } from '@testing-library/react'

import AutocompleteInput from './index'

describe('AutocompleteInput', () => {
  it('should render with a small size bar', () => {
    const options = {
      onSelect: () => `''`,
      loading: false,
      value: [],
      size: 'small',
    }

    const input = {
      onChange: () => `''`,
      onSearch: () => `''`,
      onClear: () => `''`,
      placeholder: '',
      value: '',
    }

    const { asFragment } = render(
      <AutocompleteInput input={input} options={options} />
    )

    const result = asFragment()

    expect(result).toMatchSnapshot()
  })

  it('should render with a 300px max height', () => {
    const options = {
      onSelect: () => `''`,
      loading: false,
      value: [],
      maxHeight: 300,
    }

    const input = {
      onChange: () => `''`,
      onSearch: () => `''`,
      onClear: () => `''`,
      placeholder: '',
      value: '',
    }

    const { asFragment } = render(
      <AutocompleteInput input={input} options={options} />
    )

    const result = asFragment()

    expect(result).toMatchSnapshot()
  })

  it('should render with a regular size bar', () => {
    const options = {
      onSelect: () => `''`,
      loading: false,
      value: [],
      size: 'regular',
    }

    const input = {
      onChange: () => `''`,
      onSearch: () => `''`,
      onClear: () => `''`,
      placeholder: '',
      value: '',
    }

    const { asFragment } = render(
      <AutocompleteInput input={input} options={options} />
    )

    const result = asFragment()

    expect(result).toMatchSnapshot()
  })

  it('should render with a large size bar', () => {
    const options = {
      onSelect: () => `''`,
      loading: false,
      value: [],
      size: 'large',
    }

    const input = {
      onChange: () => `''`,
      onSearch: () => `''`,
      onClear: () => `''`,
      placeholder: '',
      value: '',
    }

    const { asFragment } = render(
      <AutocompleteInput input={input} options={options} />
    )

    const result = asFragment()

    expect(result).toMatchSnapshot()
  })

  it('should render with a regular size bar when prop is absent', () => {
    const options = {
      onSelect: () => `''`,
      loading: false,
      value: [],
    }

    const input = {
      onChange: () => `''`,
      onSearch: () => `''`,
      onClear: () => `''`,
      placeholder: '',
      value: '',
    }

    const { asFragment } = render(
      <AutocompleteInput input={input} options={options} />
    )

    const result = asFragment()

    expect(result).toMatchSnapshot()
  })

  it('should render a regular version of search bar if size prop isnt small, regular or large', () => {
    const options = {
      onSelect: () => `''`,
      loading: false,
      value: [],
      size: 'regular',
    }

    const input = {
      onChange: () => `''`,
      onSearch: () => `''`,
      onClear: () => `''`,
      placeholder: '',
      value: '',
    }

    const { asFragment } = render(
      <AutocompleteInput input={input} options={options} />
    )

    const result = asFragment()

    expect(result).toMatchSnapshot()
  })
})
