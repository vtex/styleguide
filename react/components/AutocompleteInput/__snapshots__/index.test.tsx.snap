// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteInput should render a regular version of search bar if size prop isnt small, regular or large 1`] = `
<DocumentFragment>
  <div
    class="flex flex-column w-100"
  >
    <div
      class="flex flex-row"
    >
      <label
        class="relative w-100"
      >
        <div
          class="flex flex-row items-stretch overflow-hidden bw1 br2 b--solid b--muted-4 hover-b--muted-3 bg-base c-on-base br--left h-regular"
        >
          <input
            class="ma0 border-box w-100 bn outline-0 b--muted-4 hover-b--muted-3 bg-base c-on-base pr5 pl5"
            placeholder=""
            value=""
          />
        </div>
      </label>
      <button
        class="b--muted-4 hover-b--muted-3 bg-base c-on-base bg-base br2 br--right w3 bw1 ba pa0 bl-0 c-link pointer"
      >
        <svg
          class="vtex__icon-search  "
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M15.707 13.293L13 10.586C13.63 9.536 14 8.311 14 7C14 3.14 10.859 0 7 0C3.141 0 0 3.14 0 7C0 10.86 3.141 14 7 14C8.312 14 9.536 13.631 10.586 13L13.293 15.707C13.488 15.902 13.744 16 14 16C14.256 16 14.512 15.902 14.707 15.707L15.707 14.707C16.098 14.316 16.098 13.684 15.707 13.293ZM7 12C4.239 12 2 9.761 2 7C2 4.239 4.239 2 7 2C9.761 2 12 4.239 12 7C12 9.761 9.761 12 7 12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="relative"
    />
  </div>
</DocumentFragment>
`;

exports[`AutocompleteInput should render with a 300px max height 1`] = `
<DocumentFragment>
  <div
    class="flex flex-column w-100"
  >
    <div
      class="flex flex-row"
    >
      <label
        class="relative w-100"
      >
        <div
          class="flex flex-row items-stretch overflow-hidden bw1 br2 b--solid b--muted-4 hover-b--muted-3 bg-base c-on-base br--left h-regular"
        >
          <input
            class="ma0 border-box w-100 bn outline-0 b--muted-4 hover-b--muted-3 bg-base c-on-base pr5 pl5"
            placeholder=""
            value=""
          />
        </div>
      </label>
      <button
        class="b--muted-4 hover-b--muted-3 bg-base c-on-base bg-base br2 br--right w3 bw1 ba pa0 bl-0 c-link pointer"
      >
        <svg
          class="vtex__icon-search  "
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M15.707 13.293L13 10.586C13.63 9.536 14 8.311 14 7C14 3.14 10.859 0 7 0C3.141 0 0 3.14 0 7C0 10.86 3.141 14 7 14C8.312 14 9.536 13.631 10.586 13L13.293 15.707C13.488 15.902 13.744 16 14 16C14.256 16 14.512 15.902 14.707 15.707L15.707 14.707C16.098 14.316 16.098 13.684 15.707 13.293ZM7 12C4.239 12 2 9.761 2 7C2 4.239 4.239 2 7 2C9.761 2 12 4.239 12 7C12 9.761 9.761 12 7 12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="relative"
    />
  </div>
</DocumentFragment>
`;

exports[`AutocompleteInput should render with a large size bar 1`] = `
<DocumentFragment>
  <div
    class="flex flex-column w-100"
  >
    <div
      class="flex flex-row"
    >
      <label
        class="relative w-100"
      >
        <div
          class="flex flex-row items-stretch overflow-hidden bw1 br2 b--solid b--muted-4 hover-b--muted-3 bg-base c-on-base br--left h-large"
        >
          <input
            class="ma0 border-box w-100 bn outline-0 b--muted-4 hover-b--muted-3 bg-base c-on-base pr5 pl5"
            placeholder=""
            value=""
          />
        </div>
      </label>
      <button
        class="b--muted-4 hover-b--muted-3 bg-base c-on-base bg-base br2 br--right w3 bw1 ba pa0 bl-0 c-link pointer"
      >
        <svg
          class="vtex__icon-search  "
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M15.707 13.293L13 10.586C13.63 9.536 14 8.311 14 7C14 3.14 10.859 0 7 0C3.141 0 0 3.14 0 7C0 10.86 3.141 14 7 14C8.312 14 9.536 13.631 10.586 13L13.293 15.707C13.488 15.902 13.744 16 14 16C14.256 16 14.512 15.902 14.707 15.707L15.707 14.707C16.098 14.316 16.098 13.684 15.707 13.293ZM7 12C4.239 12 2 9.761 2 7C2 4.239 4.239 2 7 2C9.761 2 12 4.239 12 7C12 9.761 9.761 12 7 12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="relative"
    />
  </div>
</DocumentFragment>
`;

exports[`AutocompleteInput should render with a regular size bar 1`] = `
<DocumentFragment>
  <div
    class="flex flex-column w-100"
  >
    <div
      class="flex flex-row"
    >
      <label
        class="relative w-100"
      >
        <div
          class="flex flex-row items-stretch overflow-hidden bw1 br2 b--solid b--muted-4 hover-b--muted-3 bg-base c-on-base br--left h-regular"
        >
          <input
            class="ma0 border-box w-100 bn outline-0 b--muted-4 hover-b--muted-3 bg-base c-on-base pr5 pl5"
            placeholder=""
            value=""
          />
        </div>
      </label>
      <button
        class="b--muted-4 hover-b--muted-3 bg-base c-on-base bg-base br2 br--right w3 bw1 ba pa0 bl-0 c-link pointer"
      >
        <svg
          class="vtex__icon-search  "
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M15.707 13.293L13 10.586C13.63 9.536 14 8.311 14 7C14 3.14 10.859 0 7 0C3.141 0 0 3.14 0 7C0 10.86 3.141 14 7 14C8.312 14 9.536 13.631 10.586 13L13.293 15.707C13.488 15.902 13.744 16 14 16C14.256 16 14.512 15.902 14.707 15.707L15.707 14.707C16.098 14.316 16.098 13.684 15.707 13.293ZM7 12C4.239 12 2 9.761 2 7C2 4.239 4.239 2 7 2C9.761 2 12 4.239 12 7C12 9.761 9.761 12 7 12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="relative"
    />
  </div>
</DocumentFragment>
`;

exports[`AutocompleteInput should render with a regular size bar when prop is absent 1`] = `
<DocumentFragment>
  <div
    class="flex flex-column w-100"
  >
    <div
      class="flex flex-row"
    >
      <label
        class="relative w-100"
      >
        <div
          class="flex flex-row items-stretch overflow-hidden bw1 br2 b--solid b--muted-4 hover-b--muted-3 bg-base c-on-base br--left h-regular"
        >
          <input
            class="ma0 border-box w-100 bn outline-0 b--muted-4 hover-b--muted-3 bg-base c-on-base pr5 pl5"
            placeholder=""
            value=""
          />
        </div>
      </label>
      <button
        class="b--muted-4 hover-b--muted-3 bg-base c-on-base bg-base br2 br--right w3 bw1 ba pa0 bl-0 c-link pointer"
      >
        <svg
          class="vtex__icon-search  "
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M15.707 13.293L13 10.586C13.63 9.536 14 8.311 14 7C14 3.14 10.859 0 7 0C3.141 0 0 3.14 0 7C0 10.86 3.141 14 7 14C8.312 14 9.536 13.631 10.586 13L13.293 15.707C13.488 15.902 13.744 16 14 16C14.256 16 14.512 15.902 14.707 15.707L15.707 14.707C16.098 14.316 16.098 13.684 15.707 13.293ZM7 12C4.239 12 2 9.761 2 7C2 4.239 4.239 2 7 2C9.761 2 12 4.239 12 7C12 9.761 9.761 12 7 12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="relative"
    />
  </div>
</DocumentFragment>
`;

exports[`AutocompleteInput should render with a small size bar 1`] = `
<DocumentFragment>
  <div
    class="flex flex-column w-100"
  >
    <div
      class="flex flex-row"
    >
      <label
        class="relative w-100"
      >
        <div
          class="flex flex-row items-stretch overflow-hidden bw1 br2 b--solid b--muted-4 hover-b--muted-3 bg-base c-on-base br--left h-small"
        >
          <input
            class="ma0 border-box w-100 bn outline-0 b--muted-4 hover-b--muted-3 bg-base c-on-base pr5 pl5"
            placeholder=""
            value=""
          />
        </div>
      </label>
      <button
        class="b--muted-4 hover-b--muted-3 bg-base c-on-base bg-base br2 br--right w3 bw1 ba pa0 bl-0 c-link pointer"
      >
        <svg
          class="vtex__icon-search  "
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M15.707 13.293L13 10.586C13.63 9.536 14 8.311 14 7C14 3.14 10.859 0 7 0C3.141 0 0 3.14 0 7C0 10.86 3.141 14 7 14C8.312 14 9.536 13.631 10.586 13L13.293 15.707C13.488 15.902 13.744 16 14 16C14.256 16 14.512 15.902 14.707 15.707L15.707 14.707C16.098 14.316 16.098 13.684 15.707 13.293ZM7 12C4.239 12 2 9.761 2 7C2 4.239 4.239 2 7 2C9.761 2 12 4.239 12 7C12 9.761 9.761 12 7 12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="relative"
    />
  </div>
</DocumentFragment>
`;
